### ISSUE TITLE
Support NuGet `dotnet-tools.json` file in the root of the project

### ISSUE THREAD

**Issue by latonz** [dependabot/dependabot-core#7969]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

Since https://github.com/dependabot/dependabot-core/issues/1697 / https://github.com/dependabot/dependabot-core/pull/7028 local .NET tool files are supported. NET local tool files (`dotnet-tools.json`) can be stored in the root of a dotnet project or in `.config`. Currently only files stored in `.config` are supported. It would be nice if `dotnet-tools.json` files stored in the root of a project were also supported.

**Comment by mburumaxwell** [comment]
The default and the recommendation is to keep the `dotnet-tools.json` file in the `.config` folder. See https://github.com/dotnet/sdk/issues/9977

Could you explain under which circumstances the file would be located elsewhere? Is it for the root (also disabled by default -> https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/manifest-search)?

**Comment by latonz** [comment]
Oh I didn't know storing the file in the solution root is deprecated, thanks for pointing out! However, there are still many projects storing it in the root, therefore it is probably still a good idea to support it. But I can understand if Dependabot doesn't add support for deprecated features.

**Comment by deivid-rodriguez** [comment]
Is `dotnet-tools.json` in project root really deprecated? My understanding of https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/manifest-search is that it's unrelated to  `.` vs `./.config`, but to whether to go search up until `C:/` or not.

**Comment by alexrp** [comment]
> Is `dotnet-tools.json` in project root really deprecated? My understanding of https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/manifest-search is that it's unrelated to `.` vs `./.config`, but to whether to go search up until `C:/` or not.

This is correct. I still have `dotnet-tools.json` in the root of all of my .NET repositories and that has worked fine from .NET Core 3.1 all the way up to .NET 8.

**Comment by Frulfump** [comment]
Still relevant. Also seems like it being placed in sub-folders(in .config, related: https://github.com/dependabot/dependabot-core/issues/12910) isn't working now either.

@brettfo @JamieMagee
